client/src/components/mapping/ConceptMap/index.tsx

```diff
@@ -0,0 +1,569 @@
+              setConceptMapTitle(target.value);
+            }}
+            placeholder="Concept map title..."
+            value={conceptMapTitle}
+            small={true}
+          />
+        ) : (
+          conceptMapTitle
+        )}
+      </div>
+      <div className="row">
+        <h2 className="left">Name</h2>
+        {!existingConceptMapId || modifyAnyway ? (
+          <InputGroup
+            className="right"
+            onChange={(e: React.FormEvent<HTMLElement>): void => {
+              const target = e.target as HTMLInputElement;
+              setConceptMapName(target.value);
+            }}
+            placeholder="Concept map name..."
+            value={conceptMapName}
+            small={true}
+          />
+        ) : (
+          conceptMapName
+        )}
+      </div>
+      <div className="row">
+        <h2 className="left">Description</h2>
+        {!existingConceptMapId || modifyAnyway ? (
+          <InputGroup
+            className="right"
+            onChange={(e: React.FormEvent<HTMLElement>): void => {
+              const target = e.target as HTMLInputElement;
+              setConceptMapDescription(target.value);
+            }}
+            placeholder="Concept map description..."
+            value={conceptMapDescription}
+            small={true}
+          />
+        ) : (
+          conceptMapDescription
+        )}
+      </div>
+    </div>
+  );
+
+  const onSubmit = async (
+    e: React.FormEvent<HTMLFormElement>
+  ): Promise<void> => {
+    e.preventDefault();
+    if (creatingNewCodeSystem) {
+      const codeSystem = createCodeSystem();
+      try {
+        await axios.post(`${FHIR_API_URL}/CodeSystem`, codeSystem);
+        console.log('created code system', codeSystem);
+      } catch (err) {
+        console.error(
+          `Could not create CodeSystem: ${
+            err.response ? err.response.data : err.message
+          }`
+        );
+      }
+    }
+    if (!existingConceptMapId) {
+      const conceptMap = createConceptMap();
+      try {
+        await axios.post(`${FHIR_API_URL}/ConceptMap`, conceptMap);
+        console.log('created concept map', conceptMap);
+      } catch (err) {
+        console.error(
+          `Could not create ConceptMap: ${
+            err.response ? err.response.data : err.message
+          }`
+        );
+      }
+    } else if (modifyAnyway) {
+      const conceptMap = createConceptMap();
+      // TODO doesn't work
+      try {
+        await axios.put(
+          `${FHIR_API_URL}/ConceptMap/${existingConceptMapId}`,
+          conceptMap
+        );
+        console.log('updated concept map', conceptMap);
+      } catch (err) {
+        console.error(
+          `Could not update ConceptMap: ${
+            err.response ? err.response.data : err.message
+          }`
+        );
+      }
+    }
+    updateInputCallback(conceptMapName);
+  };
+
+  useEffect(() => {
+    let existingConceptMapGroup: Group | undefined = undefined;
```

Comment:
you don't need to assign it to undefined, do you ?

Reply:
That's right